### PR TITLE
feat: add optional dependency to collect garbage collector metrics too

### PR DESCRIPTION
**Pull request by matteodisabatino**
It could be useful to collect garbage collector metrics too, so I decided to create this PR. The dependency is marked as optional.

### CODE REVIEW

**Comment by matteodisabatino** [comment]
@joao-fontenele any news on that?

**Comment by joao-fontenele** [comment]
hello @matteodisabatino I'll look into it at some point this week

File: src/index.js [hunk 1]
```diff
@@ -65,6 +66,23 @@ module.exports = (userOptions = {}) => {
...
+    // if the option has been turned on, we start collecting garbage
+    // collector metrics too. using try/catch because the dependency is
+    // optional and it could not be installed
+    try {
+      /* eslint-disable global-require */
+      /* eslint-disable import/no-extraneous-dependencies */
+      const gcStats = require('prometheus-gc-stats');
+      /* eslint-enable import/no-extraneous-dependencies */
+      /* eslint-enable global-require */
+      const startGcStats = gcStats(Prometheus.register);
```
Comment (joao-fontenele) [inline src/index.js:79]: please use `options.prefix` here too

File: package.json [hunk 2]
```diff
@@ -28,5 +28,8 @@
     "prom-client": "^11.1.1",
     "response-time": "^2.3.2",
     "url-value-parser": "^2.0.0"
+  },
+  "optionalDependencies": {
+    "prometheus-gc-stats": "^0.6.2"
```
Comment (joao-fontenele) [inline package.json:33]: it seems you didn't update yarn.lock file together with this change

**Comment by matteodisabatino** [comment]
@joao-fontenele I think it's all done... there are just two problems:
1. I don't know how to solve this conflict
2. I don't know how to squash a commits

**Comment by joao-fontenele** [comment]
hello @matteodisabatino 

I've squashed the commits and resolved the conflict in #26 is it alright if I merge it there, or will you try to fix it?

**Comment by matteodisabatino** [comment]
@joao-fontenele no problem for me. The important is close the PR :-). After you merge have I to close this one

**Comment by joao-fontenele** [comment]
PR #26 was merged and a new release was generated

**Pull request closed without merging**